text/0000-is-sorted.md

```diff
@@ -0,0 +1,296 @@
+- Feature Name: is_sorted
+- Start Date: 2018-02-24
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add the methods `is_sorted`, `is_sorted_by` and `is_sorted_by_key` to `[T]`;
+add the methods `is_sorted` and `is_sorted_by` to `Iterator`.
+
+# Motivation
+[motivation]: #motivation
+
+In quite a few situations, one needs to check whether a sequence of elements
+is sorted. The most important use cases are probably **unit tests** and
+**pre-/post-condition checks**.
+
+The lack of an `is_sorted()` function in Rust's standard library has led to
+[countless programmers implementing their own](https://github.com/search?l=Rust&q=%22fn+is_sorted%22&type=Code&utf8=%E2%9C%93).
+While it is possible to write a one-liner using iterators (e.g.
+`(0..arr.len() - 1).all(|i| arr[i] < arr[i + 1])`), it is still unnecessary
```

Comment:
This should be a `<=` (thanks @orlp)

Reply:
That was my secret plan: hide a subtle bug in the "motivation" section that no one notices for months. Now that it has been discovered, it's the perfect motivation that `is_sorted` is certainly needed! Muhaha!

In all seriousness, it happens [all](https://stackoverflow.com/posts/51272639/revisions) [the](https://github.com/rust-lang/rfcs/pull/2351#issuecomment-370126518) [time](https://github.com/rust-lang/rfcs/pull/2351#discussion_r201497580).

<sup>Will fix this bug in the RFC later. It's blocked right now anyway.</sup>

Reply:
Ping, given FCP :)

Reply:
Done.

Reply:
Heh, I'd almost love to see a section in the motivation showcasing just how easy it is to make this mistake!

Reply:
@ExpHP Great idea, done!